src/extension/executors/runner.ts

```diff
@@ -119,15 +119,29 @@ export async function executeRunner(
     program.handleInput(input)
   })
 
-  const useIntegrated = isRunmeIntegratedTerminalEnabled()
+  const revealNotebookTerminal = background ?
+    isNotebookTerminalFeatureEnabled('backgroundTask') :
+    isNotebookTerminalFeatureEnabled('interactive')
```

Comment:
```suggestion
  const revealNotebookTerminal = interactive ?
    background ?
      isNotebookTerminalFeatureEnabled('backgroundTask') :
      isNotebookTerminalFeatureEnabled('interactive') :
    isNotebookTerminalFeatureEnabled('nonInteractive')
    
  if (revealNotebokTerminal) {
      await replaceOutput(exec, [
        new NotebookCellOutput([
          NotebookCellOutputItem.json(NotebookTerminal.create(cellUUID), OutputType.terminal),
        ]),
      ])
    }
```

Why not do it like this? Then you can remove all subsequent terminal `replaceOutput` calls, and replace checks to `isNotebookTerminalFeatureEnabled` with `revealNotebookTerminal`

Reply:
Good suggestion, fixed!